src/arrayMethodJoin.js

```diff
@@ -4,8 +4,35 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    const arr = [...this];
+    let str = '';
+
+    for (let i = 0; i < arr.length; i++) {
+      switch (arr[i]) {
+        case undefined:
+          arr[i] = '';
+          break;
+        case NaN:
+          arr[i] = 'NaN';
+          break;
+        case false:
+          arr[i] = 'false';
+          break;
+        case null:
+          arr[i] = '';
+          break;
+      }
+
+      if (i === 0) {
+        str = str + arr[i];
+      } else {
+        str = str + separator + arr[i];
+      }
```

Comment:
You need only 2 cases
1. arr[i] !== undefined && arr[i] !== null - to exclude it from array
2. i !== this.length - 1 - to disable separator adding after last element

That's all
Improve your solution and remove redundant code

Reply:
2, I don't use after I use before delimiter